test/reg-test/fam-api-reg/fam_context_mt_reg_test.cpp

```diff
@@ -0,0 +1,337 @@
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+ * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * See https://spdx.org/licenses/BSD-3-Clause
+ *
+ */
+/* Test Case Description: Tests fam_context_open and fam_context_close
+ * operations for multithreaded model.
+ */
+
+#include <fam/fam.h>
+#include <fam/fam_exception.h>
+#include <gtest/gtest.h>
+#include <iostream>
+#include <limits.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "common/fam_test_config.h"
+
+using namespace std;
+using namespace openfam;
+fam *my_fam;
+Fam_Options fam_opts;
+
+Fam_Region_Descriptor *testRegionDesc;
+const char *testRegionStr;
+int rc;
+#define NUM_THREADS 10
+#define REGION_SIZE (32 * 1024 * NUM_THREADS)
+#define REGION_PERM 0777
+#define NUM_ITERATIONS 100
+#define NUM_IO_ITERATIONS 5
+#define NUM_CONTEXTS 256
+typedef struct {
+    Fam_Descriptor *item;
+    uint64_t offset;
+    int32_t tid;
+    int32_t deltaValue;
+} ValueInfo;
+
+void *thrd_fam_context_mt(void *arg) {
+    fam_context *ctx;
+    EXPECT_NO_THROW(ctx = my_fam->fam_context_open());
+    EXPECT_NO_THROW(my_fam->fam_context_close(ctx));
+    pthread_exit(NULL);
+}
+
+void *thrd_fam_context_mt_stress(void *arg) {
+    fam_context *ctx;
+    for (int i = 0; i < NUM_ITERATIONS; i++) {
+        EXPECT_NO_THROW(ctx = my_fam->fam_context_open());
+        EXPECT_NO_THROW(my_fam->fam_context_close(ctx));
+    }
+    pthread_exit(NULL);
+}
+
+void *thrd_fam_context_mt_with_io(void *arg) {
+
+    ValueInfo *addInfo = (ValueInfo *)arg;
+    Fam_Descriptor *item = addInfo->item;
+    fam_context *ctx;
+    EXPECT_NO_THROW(ctx = my_fam->fam_context_open());
+    EXPECT_NO_THROW(ctx->fam_or(item, 0, (uint32_t)1));
+    EXPECT_NO_THROW(ctx->fam_or(item, 0, (uint32_t)2));
+    EXPECT_NO_THROW(ctx->fam_or(item, 0, (uint32_t)3));
+    EXPECT_NO_THROW(ctx->fam_or(item, 0, (uint32_t)4));
+    EXPECT_NO_THROW(my_fam->fam_or(item, 0, (uint32_t)4));
+    EXPECT_NO_THROW(ctx->fam_quiet());
+    EXPECT_NO_THROW(my_fam->fam_quiet());
+    // We are done with the operations. Destroy the region and everything in it

```

Comment:
Can you remove second part of the comment. We are not destroying the region here.